packages/firestore/src/remote/remote_store.ts

```diff
@@ -275,12 +295,22 @@ export class RemoteStore implements TargetMetadataProvider {
    */
   private shouldStartWatchStream(): boolean {
     return (
-      this.isNetworkEnabled() &&
-      !this.watchStream.isStarted() &&
+      this.shouldStartWatchStreamAfterError() &&
       !objUtils.isEmpty(this.listenTargets)
     );
   }
 
+  /**
+   * Returns whether the watch stream should be started after an error. Similar
+   * to {@link shouldStartWatchStream} except this one does not check whether
+   * there are active listen targets. For certain cases, even there is no
+   * listener, we may still want to keep the watch stream live, see
+   * https://github.com/firebase/firebase-ios-sdk/issues/1165.
+   */
+  private shouldStartWatchStreamAfterError(): boolean {
+    return this.isNetworkEnabled() && !this.watchStream.isStarted();
```

Comment:
Reading this code and the comments it would _appear_ that we'll just restart the stream forever...  but in fact the idle timeout should eventually fire with a non-error onClose() call and we'll stop restarting the stream.  At minimum we should explain this in the comments so somebody can reason about how the stream will end up being closed...